urlutil.py

```diff
@@ -26,3 +31,18 @@ def get_local_file_from_url(url, tempdir):
             output.write(stream.read())
 
     return pathname
+
+def find_firmware_by_hash(hash, board, language):
+    resource = boto3.resource('s3')
+    resource.meta.client.meta.events.register('choose-signer.s3.*', disable_signing)
+    bucket = resource.Bucket('adafruit-circuit-python')
+
+    for item in bucket.objects.filter(Prefix="bin/{}/{}/".format(board, language)):
```

Comment:
So I dorked around with this a bit and I can see the URL https://adafruit-circuit-python.s3.amazonaws.com/index.html?prefix=bin/feather_m0_rfm69/en_US/ has what we need in it. Can we use that instead of Boto3? For some reason the S3 bucket futzing is making me sad.

Reply:
I see now that this is a bunch of client JS code and not helpful